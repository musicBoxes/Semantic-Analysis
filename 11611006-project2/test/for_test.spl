int for_test()
{
	int i = 0;
	for (int j = 0 ; j < 10 ; j = j + 1) {  }
	for ( ; ; ) {  }
	for ( ; k < 10 ; k = k + 1)
	{
		k = k * 100;
	}
	for (i = 1 ; i < 2 ; i = i + 1)
	{
		i = i * 10;
	}
	return i;
}