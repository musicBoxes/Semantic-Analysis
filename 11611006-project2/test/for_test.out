Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: for_test
        LP
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: i
                ASSIGN
                Exp (3)
                  INT: 0
            SEMI
        StmtList (4)
          Stmt (4)
            FOR
            LP
            Def (4)
              Specifier (4)
                TYPE: int
              DecList (4)
                Dec (4)
                  VarDec (4)
                    ID: j
                  ASSIGN
                  Exp (4)
                    INT: 0
              SEMI
            ExpList (4)
              Exp (4)
                Exp (4)
                  ID: j
                LT
                Exp (4)
                  INT: 10
            SEMI
            ExpList (4)
              Exp (4)
                Exp (4)
                  ID: j
                ASSIGN
                Exp (4)
                  Exp (4)
                    ID: j
                  PLUS
                  Exp (4)
                    INT: 1
            RP
            Stmt (4)
              CompSt (4)
                LC
                RC
          StmtList (5)
            Stmt (5)
              FOR
              LP
              SEMI
              SEMI
              RP
              Stmt (5)
                CompSt (5)
                  LC
                  RC
            StmtList (6)
              Stmt (6)
                FOR
                LP
                SEMI
                ExpList (6)
                  Exp (6)
                    Exp (6)
                      ID: k
                    LT
                    Exp (6)
                      INT: 10
                SEMI
                ExpList (6)
                  Exp (6)
                    Exp (6)
                      ID: k
                    ASSIGN
                    Exp (6)
                      Exp (6)
                        ID: k
                      PLUS
                      Exp (6)
                        INT: 1
                RP
                Stmt (7)
                  CompSt (7)
                    LC
                    StmtList (8)
                      Stmt (8)
                        Exp (8)
                          Exp (8)
                            ID: k
                          ASSIGN
                          Exp (8)
                            Exp (8)
                              ID: k
                            MUL
                            Exp (8)
                              INT: 100
                        SEMI
                    RC
              StmtList (10)
                Stmt (10)
                  FOR
                  LP
                  ExpList (10)
                    Exp (10)
                      Exp (10)
                        ID: i
                      ASSIGN
                      Exp (10)
                        INT: 1
                  SEMI
                  ExpList (10)
                    Exp (10)
                      Exp (10)
                        ID: i
                      LT
                      Exp (10)
                        INT: 2
                  SEMI
                  ExpList (10)
                    Exp (10)
                      Exp (10)
                        ID: i
                      ASSIGN
                      Exp (10)
                        Exp (10)
                          ID: i
                        PLUS
                        Exp (10)
                          INT: 1
                  RP
                  Stmt (11)
                    CompSt (11)
                      LC
                      StmtList (12)
                        Stmt (12)
                          Exp (12)
                            Exp (12)
                              ID: i
                            ASSIGN
                            Exp (12)
                              Exp (12)
                                ID: i
                              MUL
                              Exp (12)
                                INT: 10
                          SEMI
                      RC
                StmtList (14)
                  Stmt (14)
                    RETURN
                    Exp (14)
                      ID: i
                    SEMI
        RC
